### PR TITLE
modulesync 2.1.0 & add puppet 6 support

### PR DESCRIPTION
**Pull request by bastelfreak**

### CODE REVIEW

File: metadata.json [hunk 1]
```diff
@@ -27,12 +27,16 @@
     {
       "name": "puppetlabs/stdlib",
       "version_requirement": ">= 4.13.1 < 6.0.0"
+    },
+    {
+      "name": "puppetlabs/selinux_core",
```
Comment (alexjfisher) [inline metadata.json]: I think this is wrong. The module is only marked as puppet >= 6 compatible.

Comment (oranenj) [reply metadata.json]: hm. It is a dependency though :/ Puppet's metadata format is not expressive enough in this case.

I think in order to maintain backwards compatibility, we'll just need to note in the readme that selinux_core is needed for Puppet >= 6.0.0

**Comment by oranenj** [comment]
I did a bit of testing using my test puppet master.

I upgraded it to Puppet 6 and then added selinux_core to an environment containing puppet 5 agents. It seems the puppet 5 agents either ignore or do not mind that they get synced the extra providers; selmodule seems to continue working as previously.

Therefore it seems safe to include the dependency on selinux_core in the metadata file, though I'm still undecided as to whether we should.

File: metadata.json (hunk 1 above)
Comment (alexjfisher) [reply metadata.json]: That *shouldn't* be necessary.  If your puppet-agent doesn't automatically come with it, something has gone wrong.

**Comment by oranenj** [comment]
https://puppet.com/docs/puppet/6.0/type.html#supported-types-in-modules states that selmodule is included in the puppet 6 distribution. Therefore having it as a dependency is redundant, since it is already implied by having Puppet installed.

**Comment by alexjfisher** [comment]
@oranenj What we're not 100% sure on is the implications of putting it in `.fixtures.yml`.  It appears to be required for puppet 6, and currently not break puppet 5, but it still feels wrong/dangerous loading code that isn't supposed to be there into a puppet 4/5 master's environment.  If it works today, will it work tomorrow?

I'd still like to wait a bit longer for some guidance to come back from puppet. ping @binford2k ;)

Maybe puppet will ship the core modules as gems??  Maybe they'll update puppetlabs_spec_helper??

File: Gemfile [hunk 2]
```diff
@@ -11,8 +11,7 @@ def location_for(place, fake_version = nil)
 end
 
 group :test do
-  gem 'puppetlabs_spec_helper', '~> 2.6',                           :require => false
-  gem 'rspec-puppet', '~> 2.5',                                     :require => false
+  gem 'puppetlabs_spec_helper', '>= 2.10.0',                        :require => false
```
Comment (alexjfisher) [inline Gemfile]: 2.11.0 has the change needed for .fixtures

**Pull request merged**